File: AndroidCaseStudy/Fundamental/LifecycleActivityFragment/androidactivityfragment_one/src/main/java/com/algokelvin/lifecycle/fragment/OneFragment.java
package com.algokelvin.lifecycle.fragment;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.algokelvin.lifecycle.utils.OnDataPass;
import com.algokelvin.lifecycle.R;

import org.jetbrains.annotations.NotNull;

public class OneFragment extends Fragment {
    private OnDataPass onDataPass;
    private String data1;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_one, container, false);
    }

    @Override
    public void onAttach(@NonNull @org.jetbrains.annotations.NotNull Context context) {
        super.onAttach(context);
        onDataPass = (OnDataPass) context;
    }

    @Override
    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        TextView txtFragment = view.findViewById(R.id.txt_fragment_one);
        this.data1 = txtFragment.getText().toString();
//        passData(data1);

    }

    public void passData() {
        onDataPass.onDataPass(data1);
    }

}

File: AndroidFundamental/AndroidFragment/app/src/main/java/algokelvin/app/fragmentapps/OneFragment.kt
package algokelvin.app.fragmentapps

import algokelvin.app.fragmentapps.databinding.FragmentOneBinding
import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

class OneFragment : Fragment() {
    private lateinit var binding: FragmentOneBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentOneBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.i("fragment-view", "ViewCreated 1 is start")
        binding.main()
    }

    private fun FragmentOneBinding.main() {
        btnTwoFromFragmentOne.setOnClickListener {
            startActivity(Intent(requireContext(), TwoActivity::class.java))
        }
    }

    override fun onResume() {
        super.onResume()
        Log.i("fragment-resume", "Resume 1 is start")
    }

    override fun onStart() {
        super.onStart()
        Log.i("fragment-start", "Start 1 is start")
    }

    override fun onStop() {
        super.onStop()
        Log.i("fragment-stop", "Stop 1 is start")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i("fragment-destroy", "Destroy 1 is start")
    }
}